src/tiled/automapper.cpp

```diff
@@ -691,7 +691,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListNo  = false;
 
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                if (!(setLayer->contains(x + offset.x(), y + offset.y()) || setLayer->map()->infinite())) {
```

Comment:
This is what happens when I remove the check:
![screenshot from 2017-08-21 17-10-38](https://user-images.githubusercontent.com/2878908/29517690-cf1ebaae-8693-11e7-81bb-275d40e4fd7c.png)
This is what happens when I keep the check (this obviously is the expected behaviour): 
![screenshot from 2017-08-21 17-11-05](https://user-images.githubusercontent.com/2878908/29517691-d03aae02-8693-11e7-9a47-52eb45537463.png)